model/src/main/proto/exploration.proto

```diff
@@ -319,22 +336,40 @@ message HelpIndex {
   // This type is uninitialized in cases when no index is currently available (either because a hint
   // or solution hasn't yet been triggered, or because there are none to trigger).
   oneof index_type {
-    // Indicates this help index corresponds to the index of a hint within the hint list of a state.
-    int32 hint_index = 1;
+    // Indicates this help index corresponds to the index of the next available hint within the hint
+    // list of a state.
+    int32 available_next_hint_index = 1;
+
+    // Indicates this help index corresponds to the index of the last revealed hint within the hint
+    // list of a state.
+    int32 latest_revealed_hint_index = 2;
 
     // Indicates this help index corresponds to the solution of a state. The boolean value here has
     // no importance and is always 'true'.
-    bool show_solution = 2;
+    bool show_solution = 3;
 
     // Indicates that everything available has been revealed. Note that this is different than the
     // case when there are no hints or solutions to trigger, even though the resulting behavior may
     // be different. This case specifically indicates that all hints and the solution, if present,
     // have been revealed and there's no other help to provide. The boolean value here has no
     // importance and is always 'true'.
-    bool everything_revealed = 3;
+    bool everything_revealed = 4;
   }
 }
 
+// A Structure corresponding to the state of hint and solution for the current exploration or
+// training session.
+message HintState {
+  // HelpIndex for the current state.
+  HelpIndex help_index = 1;
+
+  // Sequence number of the most recently scheduled task to show help to the learner.
+  int32 hint_sequence_number = 2;
+
+  // Delay for scheduling a new task to show more help to the learner.
+  int64 delay_to_show_next_hint_and_solution = 4;
```

Comment:
```suggestion
  int64 delay_to_show_next_hint_and_solution = 3;
```

Reply:
This appears to be resolved (mainly since HintState no longer exists). Closing the thread with permission from @vinitamurthi to verify & close out old conversation threads.